src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -65,11 +69,18 @@
     @Argument(doc = "maximum number of bytes to read from a file into each partition of reads. " +
             "Setting this higher will result in fewer partitions. Note that this will not be equal to the size of the partition in memory",
             fullName = "bamPartitionSize", shortName = "bps", optional = true)
-    protected long bamPartitionSplitSize = ReadsSparkSource.DEFAULT_SPLIT_SIZE;
+    protected long bamPartitionSplitSize = 0;
 
     @Argument(fullName = "disableSequenceDictionaryValidation", shortName = "disableSequenceDictionaryValidation", doc = "If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!", optional = true)
     private boolean disableSequenceDictionaryValidation = false;
 
+    @Argument(doc = "For tools that write an output, write the output in multiple pieces (shards)", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
+    protected boolean shardedOutput = false;
+
+    @Argument(doc="For tools that shuffle data or write an output, sets the number of reducers. Defaults to one partition per 10MB of input.",
+            shortName = "P", fullName = "parallelism", optional = true)
+    protected int parallelism = 0;
```

Comment:
Make the short name of this argument `parallelism` as well, to avoid potential future collisions with tools that may have already claimed `-P`. Also, document the meaning of 0 for this argument.